libstainless/lib.rs

```diff
@@ -12,29 +12,55 @@ impl<T> Set<T> {
   pub fn empty() -> Self {
     unimplemented!()
   }
-
-  // TODO: Only take 'self' as a reference and also take the other parameters
-  //   only by reference.
   pub fn singleton(_t: &T) -> Self {
     unimplemented!()
   }
-  pub fn add(&self, _t: &T) -> Set<T> {
+
+  pub fn add(&self, _t: &T) -> Self {
+    unimplemented!()
+  }
+
+  pub fn union(&self, _other: &Set<T>) -> Self {
+    unimplemented!()
+  }
+  pub fn intersection(&self, _other: &Set<T>) -> Self {
+    unimplemented!()
+  }
+  pub fn difference(&self, _other: &Set<T>) -> Self {
+    unimplemented!()
+  }
+  pub fn is_subset_of(&self, _other: &Set<T>) -> bool {
     unimplemented!()
   }
-  pub fn contains(&self, _t: &T) -> bool {
```

Comment:
We never noticed that we don't even extract `Set::contains` because it wasn't used and didn't appear in any tests. Still this is bad, and it must be removed. (And maybe readded _with an implementation_ later.)